src/game/puppyprint.c

```diff
@@ -720,47 +880,135 @@ void render_blank_box(s32 x1, s32 y1, s32 x2, s32 y2, s32 r, s32 g, s32 b, s32 a
     gDPSetFillColor(gDisplayListHead++, (GPACK_RGBA5551(r, g, b, 1) << 16) | GPACK_RGBA5551(r, g, b, 1));
     print_set_envcolour(r, g, b, a);
     gDPFillRectangle(gDisplayListHead++, x1, y1, x2 - cycleadd, y2 - cycleadd);
+    gDPPipeSync(gDisplayListHead++);
 }
 
-extern s32 text_iterate_command(const char *str, s32 i, s32 runCMD);
-extern void get_char_from_byte(u8 letter, s32 *textX, s32 *textY, s32 *spaceX, s32 *offsetY, s32 font);
+// Same as above, but with rounded edges.
+// Follows all the same rules of usage.
+void render_blank_box_rounded(s32 x1, s32 y1, s32 x2, s32 y2, u8 r, u8 g, u8 b, u8 a) {
+    if (x2 < x1)
+    {
+        u32 temp = x2;
+        x2 = x1;
+        x1 = temp;
+    }
+    if (y2 < y1)
+    {
+        u32 temp = y2;
+        y2 = y1;
+        y1 = temp;
+    }
+    if (x1 < 0) x1 = 0;
+    if (y1 < 0) y1 = 0;
+    if (x2 > SCREEN_WIDTH) x2 = SCREEN_WIDTH;
+    if (y2 > SCREEN_HEIGHT) y2 = SCREEN_HEIGHT;
+    s32 cycleadd = 0;
+    gDPSetCycleType(gDisplayListHead++, G_CYC_1CYCLE);
+    if (a == 255) {
+        gDPSetRenderMode(gDisplayListHead++, G_RM_OPA_SURF, G_RM_OPA_SURF2);
+    } else {
+        gDPSetRenderMode(gDisplayListHead++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+    }
+    gDPSetFillColor(gDisplayListHead++, GPACK_RGBA5551(r, g, b, 1) << 16 | GPACK_RGBA5551(r, g, b, 1));
+    print_set_envcolour(r, g, b, a);
+    gDPFillRectangle(gDisplayListHead++, x1+4, y1, x2-4, y1+1);
+    gDPFillRectangle(gDisplayListHead++, x1+2, y1+1, x2-2, y1+2);
+    gDPFillRectangle(gDisplayListHead++, x1+1, y1+2, x2-1, y1+4);
+    gDPFillRectangle(gDisplayListHead++, x1+1, y2-4, x2-1, y2-2);
+    gDPFillRectangle(gDisplayListHead++, x1+2, y2-2, x2-2, y2-1);
+    gDPFillRectangle(gDisplayListHead++, x1+4, y2-1, x2-4, y2);
+    if (ABS(x1 - x2) % 4 == 0 && a == 255) {
+        gDPSetCycleType(gDisplayListHead++, G_CYC_FILL);
+        gDPSetRenderMode(gDisplayListHead++, G_RM_NOOP, G_RM_NOOP);
+        cycleadd = 1;
+    }
+    gDPFillRectangle(gDisplayListHead++, x1, y1+4, x2 - cycleadd, y2-4 - cycleadd);
+    gDPPipeSync(gDisplayListHead++);
+}
+
+s8 shakeToggle = 0;
+s8 waveToggle = 0;
+s8 rainbowToggle = 0;
+f32 textSize = 1.0f; // The value that's used as a baseline multiplier before applying text size modifiers. Make sure to set it back when you're done.
+f32 textSizeTotal = 1.0f; // The value that's read to set the text size. Do not mess with this.
+f32 textSizeTemp = 1.0f; // The value that's set when modifying text size mid draw. Also do not mess with this.
+u16 textTempScale = 1024; // A fixed point means of referring to scale.
+u8 textOffsets[2]; // Represents the dimensions of the text (12 x 8), and written to when size is modified.
+u8 topLineHeight; // Represents the peak line height of the current line. Prevents vertical overlapping.
+u8 gMonoSpace = FALSE; // Ignore kerning.
```

Comment:
these feel like struct thingymagoos but meh, problem for another day

Reply:
yeah true tbh, but meh it's been like this for who knows how long

Reply:
also typically local vars are prefixed with s, def not a blocker by any means

Reply:
I feel like that's mostly just a decomp thing, but at the very least non-externed variabled probably shouldn't be prefixed with `g`.

My assumption too is `s` is referring to `signed`, which doesn't apply to most of these values anyway.

Reply:
s means static iirc

Reply:
Oh lol

Reply:
still missing the funny `static` keyword 😔